rotkehlchen/chain/ethereum/modules/yearn/utils.py

```diff
@@ -69,29 +69,76 @@ def _maybe_reset_yearn_cache_timestamp(data: list[dict[str, Any]] | None) -> boo
     return False  # will continue
 
 
-def _query_yearn_vaults() -> tuple[list[dict[str, Any]] | None, str | None]:
-    """Query yearn v2 and v3 vaults from the ydaemon api."""
+def _query_ydaemon(endpoint: str) -> list[dict[str, Any]] | dict[str, Any] | None:
+    """Query the ydaemon api.
+    Returns the response from the api in a list or dict.
+
+    May raise:
+    - RemoteError
+    """
     timeout_tuple = CachedSettings().get_timeout_tuple()
+    url = f'{YDAEMON_API}/{endpoint}'
     try:
-        response = requests.get(YDAEMON_API, timeout=timeout_tuple)
+        response = requests.get(url, timeout=timeout_tuple)
     except requests.exceptions.RequestException as e:
-        log.error(f'Request to {YDAEMON_API} failed due to {e!s}')
-        return None, 'Failed to obtain yearn vault information'
+        log.error(f'Request to {url} failed due to {e!s}')
+        raise RemoteError('Failed to obtain yearn vault information') from e
 
     if response.status_code in (HTTPStatus.NOT_FOUND, HTTPStatus.SERVICE_UNAVAILABLE):
-        return None, 'Failed to obtain a proper response from the yearn API'
+        raise RemoteError('Failed to obtain a proper response from the yearn API')
 
     try:
         data = response.json()
     except (DeserializationError, JSONDecodeError) as e:
         log.error(f'Failed to deserialize yearn api response {response.text} as JSON due to {e!s}')
-        return None, 'Failed to deserialize data from the yearn api'
+        raise RemoteError('Failed to deserialize data from the yearn api') from e
+
+    return data
+
+
+def _query_yearn_vaults(count: int) -> list[dict[str, Any]]:
+    """Query list of yearn v2 and v3 vaults from the ydaemon api.
+    Returns the list of vaults.
+
+    Args:
+        - count (int): Total number of vaults to retrieve
+    May raise:
+        - RemoteError
+    """
+    all_vaults = []
+    page_size = 200
+    for offset in range(0, count, page_size):
+        current_limit = min(page_size, count - offset)
+        data = _query_ydaemon(f'list/vaults?chainIDs=1&limit={current_limit}&skip={offset}')
+
+        if not isinstance(data, list) or len(data) != current_limit:
```

Comment:
You don't need:

1. isinstance check for typing. Typing should be handled by making sure that `_query_ydaemon` is using proper `overload` depending on the passed endpoint. Just like we do everywhere else in the code
2. If len(data) < pagesize you just stop.